In [ ]:
Assignment 4

Using the datasets above: List_of_local_governments_in_Nigeria.csv and Nigeria_povertyrate_per_local_gov.csv
    1. Merge the dataset together to form a single table.
    2. Remove any redundant columns.
    3. Calculate how population affects poverty rate of each local government.
    4. Arrange each local government accordiing to the order of their poverty rate.
    5. Find the percentage of poverty and sort in ascending order.

In [1]:
import numpy as np
import pandas as pd

<h1>Reading Dataset for Local Governments in Nigeria</h1>

In [2]:
LG = pd.read_csv('list_of_local_government_areas_of_nigeria.csv')
LG

SNo Local Government Area  \
0                                                    1                Abadam   
1                                                    2                 Abaji   
2                                                    3                  Abak   
3                                                    4             Abakaliki   
4                                                    5             Aba North   
..                                                 ...                   ...   
771                                                772                  Zing   
772                                                773                 Zurmi   
773                                                774                  Zuru   
774                                                NaN                   NaN   
775  For more information, kindly visit :  www.down...                   NaN   

               State  Unnamed: 3  
0        Borno State         NaN  
1                FCT         NaN  
2    Akwa Ibom State         NaN  
3       Ebonyi State         NaN  
4         Abia State         NaN  
..               ...         ...  
771     Taraba State         NaN  
772    Zamfara State         NaN  
773      Kebbi State         NaN  
774              NaN         NaN  
775              NaN         NaN  

[776 rows x 4 columns]

In [3]:
LG.isnull().any()

SNo                      True
Local Government Area    True
State                    True
Unnamed: 3               True
dtype: bool

<h1>Cleaning of dataset LG</h1>

In [4]:
LG.drop(columns='Unnamed: 3', inplace=True)

<h1>Reading Datasets for Nigeria poverty rate per local government</h1>

In [19]:
povertyRate = pd.read_csv('Nigeria povertyrate per local gov.csv')
povertyRate

ID_0  ISO   NAME_0  ID_12  NAME_12  ID_23        NAME_23  2014_Popul  \
0     163  NGA  Nigeria      1     Abia      1      Aba North      257689   
1     163  NGA  Nigeria      1     Abia      2      Aba South      405791   
2     163  NGA  Nigeria      1     Abia      3       Arochukw      193250   
3     163  NGA  Nigeria      1     Abia      4          Bende      244399   
4     163  NGA  Nigeria      1     Abia      5        Ikwuano      139157   
..    ...  ...      ...    ...      ...    ...            ...         ...   
770   163  NGA  Nigeria     38  Zamfara    771           Maru      449533   
771   163  NGA  Nigeria     38  Zamfara    772       Shinkafi      220070   
772   163  NGA  Nigeria     38  Zamfara    773  Talata-Mafara      131298   
773   163  NGA  Nigeria     38  Zamfara    774          Tsafe      333706   
774   163  NGA  Nigeria     38  Zamfara    775          Zurmi      291140   

     Poverty_Ra  2014_Pover  Unnamed: 10  
0      0.017378        9661          NaN  
1      0.094480       16070          NaN  
2      0.175241       40022          NaN  
3      0.129469       39410          NaN  
4      0.111351       12145          NaN  
..          ...         ...          ...  
770    0.502184      227452          NaN  
771    0.390775       84238          NaN  
772    0.554036       72134          NaN  
773    0.565074      186716          NaN  
774    0.486278      144487          NaN  

[775 rows x 11 columns]

<h1>Cleaning of Dataset Povery rate</h1>

In [6]:
povertyRate.drop(columns= 'Unnamed: 10', inplace=True)

<h1>Merging of the two Datasets</h1>

In [7]:
newdata = pd.merge(LG, povertyRate, left_on='Local Government Area', right_on='NAME_23')
newdata

SNo Local Government Area            State  ID_0  ISO   NAME_0  ID_12  \
0      1                Abadam      Borno State   163  NGA  Nigeria      8   
1      2                 Abaji              FCT   163  NGA  Nigeria     15   
2      3                  Abak  Akwa Ibom State   163  NGA  Nigeria      3   
3      5             Aba North       Abia State   163  NGA  Nigeria      1   
4      6             Aba South       Abia State   163  NGA  Nigeria      1   
..   ...                   ...              ...   ...  ...      ...    ...   
590  769                 Zango    Katsina State   163  NGA  Nigeria     21   
591  771                 Zaria     Kaduna State   163  NGA  Nigeria     19   
592  772                  Zing     Taraba State   163  NGA  Nigeria     35   
593  773                 Zurmi    Zamfara State   163  NGA  Nigeria     38   
594  774                  Zuru      Kebbi State   163  NGA  Nigeria     22   

                       NAME_12  ID_23    NAME_23  2014_Popul  Poverty_Ra  \
0                        Borno    142     Abadam       83113    0.396480   
1    Federal Capital Territory    276      Abaji       60985    0.243414   
2                    Akwa Ibom     39       Abak      153585    0.227546   
3                         Abia      1  Aba North      257689    0.017378   
4                         Abia      2  Aba South      405791    0.094480   
..                         ...    ...        ...         ...         ...   
590                    Katsina    447      Zango      217949    0.558352   
591                     Kaduna    369      Zaria      448419    0.122632   
592                     Taraba    743       Zing      113311    0.494821   
593                    Zamfara    775      Zurmi      291140    0.486278   
594                      Kebbi    468       Zuru      159716    0.362035   

     2014_Pover  
0         31093  
1         14320  
2         33179  
3          9661  
4         16070  
..          ...  
590      117828  
591       21111  
592       56127  
593      144487  
594       51528  

[595 rows x 13 columns]

In [8]:
newdata = newdata.set_index('Local Government Area')
newdata

SNo            State  ID_0  ISO   NAME_0  ID_12  \
Local Government Area                                                    
Abadam                   1      Borno State   163  NGA  Nigeria      8   
Abaji                    2              FCT   163  NGA  Nigeria     15   
Abak                     3  Akwa Ibom State   163  NGA  Nigeria      3   
Aba North                5       Abia State   163  NGA  Nigeria      1   
Aba South                6       Abia State   163  NGA  Nigeria      1   
...                    ...              ...   ...  ...      ...    ...   
Zango                  769    Katsina State   163  NGA  Nigeria     21   
Zaria                  771     Kaduna State   163  NGA  Nigeria     19   
Zing                   772     Taraba State   163  NGA  Nigeria     35   
Zurmi                  773    Zamfara State   163  NGA  Nigeria     38   
Zuru                   774      Kebbi State   163  NGA  Nigeria     22   

                                         NAME_12  ID_23    NAME_23  \
Local Government Area                                                
Abadam                                     Borno    142     Abadam   
Abaji                  Federal Capital Territory    276      Abaji   
Abak                                   Akwa Ibom     39       Abak   
Aba North                                   Abia      1  Aba North   
Aba South                                   Abia      2  Aba South   
...                                          ...    ...        ...   
Zango                                    Katsina    447      Zango   
Zaria                                     Kaduna    369      Zaria   
Zing                                      Taraba    743       Zing   
Zurmi                                    Zamfara    775      Zurmi   
Zuru                                       Kebbi    468       Zuru   

                       2014_Popul  Poverty_Ra  2014_Pover  
Local Government Area                                      
Abadam                      83113    0.396480       31093  
Abaji                       60985    0.243414       14320  
Abak                       153585    0.227546       33179  
Aba North                  257689    0.017378        9661  
Aba South                  405791    0.094480       16070  
...                           ...         ...         ...  
Zango                      217949    0.558352      117828  
Zaria                      448419    0.122632       21111  
Zing                       113311    0.494821       56127  
Zurmi                      291140    0.486278      144487  
Zuru                       159716    0.362035       51528  

[595 rows x 12 columns]

In [9]:
newdata.drop(columns='SNo', inplace=True)

In [10]:
newdata

State  ID_0  ISO   NAME_0  ID_12  \
Local Government Area                                               
Abadam                     Borno State   163  NGA  Nigeria      8   
Abaji                              FCT   163  NGA  Nigeria     15   
Abak                   Akwa Ibom State   163  NGA  Nigeria      3   
Aba North                   Abia State   163  NGA  Nigeria      1   
Aba South                   Abia State   163  NGA  Nigeria      1   
...                                ...   ...  ...      ...    ...   
Zango                    Katsina State   163  NGA  Nigeria     21   
Zaria                     Kaduna State   163  NGA  Nigeria     19   
Zing                      Taraba State   163  NGA  Nigeria     35   
Zurmi                    Zamfara State   163  NGA  Nigeria     38   
Zuru                       Kebbi State   163  NGA  Nigeria     22   

                                         NAME_12  ID_23    NAME_23  \
Local Government Area                                                
Abadam                                     Borno    142     Abadam   
Abaji                  Federal Capital Territory    276      Abaji   
Abak                                   Akwa Ibom     39       Abak   
Aba North                                   Abia      1  Aba North   
Aba South                                   Abia      2  Aba South   
...                                          ...    ...        ...   
Zango                                    Katsina    447      Zango   
Zaria                                     Kaduna    369      Zaria   
Zing                                      Taraba    743       Zing   
Zurmi                                    Zamfara    775      Zurmi   
Zuru                                       Kebbi    468       Zuru   

                       2014_Popul  Poverty_Ra  2014_Pover  
Local Government Area                                      
Abadam                      83113    0.396480       31093  
Abaji                       60985    0.243414       14320  
Abak                       153585    0.227546       33179  
Aba North                  257689    0.017378        9661  
Aba South                  405791    0.094480       16070  
...                           ...         ...         ...  
Zango                      217949    0.558352      117828  
Zaria                      448419    0.122632       21111  
Zing                       113311    0.494821       56127  
Zurmi                      291140    0.486278      144487  
Zuru                       159716    0.362035       51528  

[595 rows x 11 columns]

<h1>To check if there are any duplicate values</h1>

In [11]:
newdata.duplicated()

Local Government Area
Abadam       False
Abaji        False
Abak         False
Aba North    False
Aba South    False
             ...  
Zango        False
Zaria        False
Zing         False
Zurmi        False
Zuru         False
Length: 595, dtype: bool

In [12]:
newdata.drop_duplicates(inplace=True)

<h1>To calculate how Population affects Poverty rate of each Local Government Area</h1>

In [13]:
# We use the pandas correlation method to get it
correlation = newdata.corr(method='spearman')
correlation

ID_0     ID_12     ID_23  2014_Popul  Poverty_Ra  2014_Pover
ID_0         NaN       NaN       NaN         NaN         NaN         NaN
ID_12        NaN  1.000000  0.999426    0.028777   -0.003250    0.022249
ID_23        NaN  0.999426  1.000000    0.027058   -0.003605    0.020865
2014_Popul   NaN  0.028777  0.027058    1.000000   -0.055736    0.440375
Poverty_Ra   NaN -0.003250 -0.003605   -0.055736    1.000000    0.812469
2014_Pover   NaN  0.022249  0.020865    0.440375    0.812469    1.000000

In [14]:
correlation.drop(index='ID_0', axis=0, inplace=True)

In [15]:
correlation.drop(columns='ID_0', inplace=True)

In [16]:
correlation

ID_12     ID_23  2014_Popul  Poverty_Ra  2014_Pover
ID_12       1.000000  0.999426    0.028777   -0.003250    0.022249
ID_23       0.999426  1.000000    0.027058   -0.003605    0.020865
2014_Popul  0.028777  0.027058    1.000000   -0.055736    0.440375
Poverty_Ra -0.003250 -0.003605   -0.055736    1.000000    0.812469
2014_Pover  0.022249  0.020865    0.440375    0.812469    1.000000

<h1>To calculate how population affects poverty rate, we need to find the correlation between both population and poverty
rate by employing the correlation method which consists of 'SPEARMAN, PEARSON AND KENDALL' after using the method and finding the
correlation we found out that population does not affect poverty because we had an inverse relationship(-0.055736), which is not between 0.5 and 1
</h1>

<h1>Sorting each local governments according to the order of their poverty rate</h1>

In [17]:
newdata.sort_values('Poverty_Ra', ascending=False)

State  ID_0  ISO   NAME_0  ID_12 NAME_12  ID_23  \
Local Government Area                                                           
Ikwo                   Ebonyi State   163  NGA  Nigeria     11  Ebonyi    218   
Ezza South             Ebonyi State   163  NGA  Nigeria     11  Ebonyi    217   
Ohaukwu                Ebonyi State   163  NGA  Nigeria     11  Ebonyi    223   
Ezza North             Ebonyi State   163  NGA  Nigeria     11  Ebonyi    216   
Ohaozara               Ebonyi State   163  NGA  Nigeria     11  Ebonyi    222   
...                             ...   ...  ...      ...    ...     ...    ...   
Bida                    Niger State   163  NGA  Nigeria     27   Niger    541   
Aba North                Abia State   163  NGA  Nigeria      1    Abia      1   
Uvwie                   Delta State   163  NGA  Nigeria     10   Delta    208   
Sokoto North           Sokoto State   163  NGA  Nigeria     34  Sokoto    720   
Fagge                    Kano State   163  NGA  Nigeria     20    Kano    381   

                            NAME_23  2014_Popul  Poverty_Ra  2014_Pover  
Local Government Area                                                    
Ikwo                           Ikwo      215257    0.914384      197512  
Ezza South               Ezza South      168892    0.903956      151608  
Ohaukwu                     Ohaukwu      253736    0.862090      220785  
Ezza North               Ezza North      158054    0.844616      131690  
Ohaozara                   Ohaozara      167245    0.842780      140108  
...                             ...         ...         ...         ...  
Bida                           Bida      240860    0.018769        5264  
Aba North                 Aba North      257689    0.017378        9661  
Uvwie                         Uvwie      289312    0.016641        4621  
Sokoto North           Sokoto North      246991    0.009919        1496  
Fagge                         Fagge      859163    0.004566        4250  

[595 rows x 11 columns]

<h1>Percentage of Poverty</h1>

In [20]:
newdata['Percentage(%)'] = newdata['Poverty_Ra']*100
newdata['Percentage(%)']

Local Government Area
Abadam       39.64800
Abaji        24.34140
Abak         22.75460
Aba North     1.73775
Aba South     9.44801
               ...   
Zango        55.83520
Zaria        12.26320
Zing         49.48210
Zurmi        48.62780
Zuru         36.20350
Name: Percentage(%), Length: 595, dtype: float64

In [25]:
newdata.sort_values('Percentage(%)', ascending=True)

State  ID_0  ISO   NAME_0  ID_12 NAME_12  ID_23  \
Local Government Area                                                           
Fagge                    Kano State   163  NGA  Nigeria     20    Kano    381   
Sokoto North           Sokoto State   163  NGA  Nigeria     34  Sokoto    720   
Uvwie                   Delta State   163  NGA  Nigeria     10   Delta    208   
Aba North                Abia State   163  NGA  Nigeria      1    Abia      1   
Bida                    Niger State   163  NGA  Nigeria     27   Niger    541   
...                             ...   ...  ...      ...    ...     ...    ...   
Ohaozara               Ebonyi State   163  NGA  Nigeria     11  Ebonyi    222   
Ezza North             Ebonyi State   163  NGA  Nigeria     11  Ebonyi    216   
Ohaukwu                Ebonyi State   163  NGA  Nigeria     11  Ebonyi    223   
Ezza South             Ebonyi State   163  NGA  Nigeria     11  Ebonyi    217   
Ikwo                   Ebonyi State   163  NGA  Nigeria     11  Ebonyi    218   

                            NAME_23  2014_Popul  Poverty_Ra  2014_Pover  \
Local Government Area                                                     
Fagge                         Fagge      859163    0.004566        4250   
Sokoto North           Sokoto North      246991    0.009919        1496   
Uvwie                         Uvwie      289312    0.016641        4621   
Aba North                 Aba North      257689    0.017378        9661   
Bida                           Bida      240860    0.018769        5264   
...                             ...         ...         ...         ...   
Ohaozara                   Ohaozara      167245    0.842780      140108   
Ezza North               Ezza North      158054    0.844616      131690   
Ohaukwu                     Ohaukwu      253736    0.862090      220785   
Ezza South               Ezza South      168892    0.903956      151608   
Ikwo                           Ikwo      215257    0.914384      197512   

                       Percentage(%)  
Local Government Area                 
Fagge                        0.45664  
Sokoto North                 0.99191  
Uvwie                        1.66406  
Aba North                    1.73775  
Bida                         1.87686  
...                              ...  
Ohaozara                    84.27800  
Ezza North                  84.46160  
Ohaukwu                     86.20900  
Ezza South                  90.39560  
Ikwo                        91.43840  

[595 rows x 12 columns]